File: Hot 100/75. 颜色分类.go
func sortColors(nums []int) {
	p0, p1 := 0, 0
	for i, num := range nums {
		if num == 1 {
			nums[i], nums[p1] = nums[p1], nums[i]
			p1++
		} else if num == 0 {
			nums[i], nums[p0] = nums[p0], nums[i]
			if p0 < p1 {
				nums[i], nums[p1] = nums[p1], nums[i]
			}
			p0++
			p1++
		}
	}
}

File: 刷题/Hot 100/75. 颜色分类.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int n=nums.size();
        int p0=0,p1=0;
        for(int i=0;i<n;i++){
            if(nums[i]==1){
                swap(nums[p1],nums[i]);
                p1++;
            }else if(nums[i]==0){
                swap(nums[p0],nums[i]);
                if(p0<p1){
                    swap(nums[p1],nums[i]);
                }
                p0++,p1++;
            }
        }
    }
};